metagraph/integration_tests/test_transform_anno.py

```diff
@@ -146,5 +146,245 @@ def test_aggregate_columns_filtered(self):
+            expected = self._compute_expected_aggregated_count(logan_30_counts, logan_30_alt_counts)
+            actual = agg_counts.get(kmer, 0)
+            
+            self.assertEqual(actual, expected,
+                f"K-mer {kmer}: expected {expected} (sum of {sum(logan_30_counts)}+{sum(logan_30_alt_counts)}), got {actual}")
+    
+    def test_aggregate_counts_with_value_filter(self):
+        """Test count aggregation with min/max-value filtering."""
+        # Filter: only count values in range [10, 100]
+        agg_counts = self._aggregate_and_query(count_width=16, min_value=10, max_value=100,
+                                               output_name='agg_value_filter')
+        
+        # Verify each test k-mer
+        for kmer, (logan_30_counts, logan_30_alt_counts) in self.test_kmers.items():
+            expected = self._compute_expected_aggregated_count(
+                logan_30_counts, logan_30_alt_counts,
+                min_value=10, max_value=100
+            )
+            actual = agg_counts.get(kmer, 0)
+            
+            self.assertEqual(actual, expected,
+                f"K-mer {kmer}: expected {expected} with value filter [10,100], got {actual}")
+    
+    def test_aggregate_counts_with_count_filter(self):
+        """Test count aggregation with min/max-count filtering."""
+        # Filter: only include k-mers with aggregated count in [5, 100]
+        agg_counts = self._aggregate_and_query(count_width=16, min_count=5, max_count=100,
+                                               output_name='agg_count_filter')
+        
+        # Verify each test k-mer
+        for kmer, (logan_30_counts, logan_30_alt_counts) in self.test_kmers.items():
+            expected = self._compute_expected_aggregated_count(
+                logan_30_counts, logan_30_alt_counts,
+                min_count=5, max_count=100
+            )
+            actual = agg_counts.get(kmer, 0)
+            
+            self.assertEqual(actual, expected,
+                f"K-mer {kmer}: expected {expected} with min_count=5, max_count=100, got {actual}")
+    
+    def test_aggregate_counts_with_width_saturation(self):
+        """Test count aggregation with reduced count-width causing saturation."""
+        # Use count-width 8 (max 255) - some counts will saturate
+        agg_counts = self._aggregate_and_query(count_width=8, output_name='agg_width8')
+        
+        # Verify each test k-mer with saturation at 255
+        for kmer, (logan_30_counts, logan_30_alt_counts) in self.test_kmers.items():
+            # Compute expected with saturation: cap each count at 255, then sum
+            saturated_30 = [min(c, 255) for c in logan_30_counts]
+            saturated_alt = [min(c, 255) for c in logan_30_alt_counts]
+            expected_raw = sum(saturated_30) + sum(saturated_alt)
+            expected = min(expected_raw, 255)  # Final result also saturates at 255
```

Comment:
are there any counts greater than 255 in the test file, actually?

Reply:
Would be good to add tests for edge cases. Even a single k-mer. Can be generated also in python.
with with different values of count-width, max_count, max_frequency, etc.